### PR TITLE
Breaking: remove needs_parentheses

### PR DESCRIPTION
**Pull request by fingolfin**
Also remove displayed_with_minus_in_front, show_minus_one

### CODE REVIEW

File: src/FreeAssAlgebra.jl [hunk 1]
```diff
@@ -67,7 +67,7 @@ end
 
 function expressify(a::FreeAssAlgebra; context = nothing)
    return Expr(:sequence, Expr(:text, "Free associative algebra over "),
-                          expressify(base_ring(a)),
+                          string(base_ring(a)),
```
Comment (fingolfin) [inline src/FreeAssAlgebra.jl:70]: This was relying on the fallback `expressify`. Does this change seem reasonable to you, @tthsqe12 ?

Comment (tthsqe12) [reply src/FreeAssAlgebra.jl:70]: This does not seems like a reasonable change. If anything, it should be
`expressify(base_ring(a), context = context)`. If expressify on the base ring broken or going to be broken soon?

Comment (thofma) [reply src/FreeAssAlgebra.jl:70]: Isn't `string` falling back to the same thing eventually?

File: src/PrettyPrinting.jl [hunk 2]
```diff
@@ -155,14 +155,7 @@ end
...
 
-# generic fallback in case expressify is not defined
-function expressify(@nospecialize(a); context = nothing)::String
-    s = sprint(print, a; context = context)::String
-    if needs_parentheses(a)::Bool
-        return "(" * s * ")"
-    else
-        return s
-    end
+function expressify
```
Comment (tthsqe12) [inline src/PrettyPrinting.jl:158]: Ah, now all is clear. This fallback should be not removed, and should be simply changed to
`return sprint(print, a; context = context)` if needs_parentheses is going away.

File: src/FreeAssAlgebra.jl (hunk 1 above)
Comment (tthsqe12) [reply src/FreeAssAlgebra.jl:70]: This pr was canning the string fallback, to which I object.

File: src/PrettyPrinting.jl [hunk 3]
```diff
@@ -155,14 +155,7 @@ end
...
 end
```
Comment (thofma) [inline src/PrettyPrinting.jl:159]:
```suggestion
   return sprint(print, a; context = context)::String
end
```

Comment (thofma) [reply src/PrettyPrinting.jl:159]: Like this @tthsqe12?

Comment (tthsqe12) [reply src/PrettyPrinting.jl:159]: sure

File: src/FreeAssAlgebra.jl (hunk 1 above)
Comment (thofma) [inline src/FreeAssAlgebra.jl:70]:
```suggestion
                          expressify(base_ring(a), context = context),
```

**Comment by thofma** [comment]
I think this is good to go. Will merge this once we have collected some breaking changes.

File: src/FreeAssAlgebra.jl (hunk 1 above)
Comment (tthsqe12) [reply src/FreeAssAlgebra.jl]: We are not canning the string fallback.

Comment (thofma) [reply src/FreeAssAlgebra.jl]: Is this an agreement or a complaint?

**Review by thofma — APPROVED** [review]

**Comment by fingolfin** [comment]
Thanks for checking and fixing this!

**Comment by thofma** [comment]
Annoyingly this needs a rebase. The important/export lines are a bit more confusing ...

**Pull request merged**